pkg/oauth/external/handler_test.go

```diff
@@ -149,3 +247,71 @@ func TestRedirectingStateError(t *testing.T) {
 		t.Errorf("Expected original error back, got %#v", err)
 	}
 }
+
+type attributes struct{}
+
+func (attributes) GetUser() user.Info      { return nil }
+func (attributes) GetVerb() string         { return "" }
+func (attributes) IsReadOnly() bool        { return true }
+func (attributes) GetNamespace() string    { return "" }
+func (attributes) GetResource() string     { return "" }
+func (attributes) GetSubresource() string  { return "" }
+func (attributes) GetName() string         { return "" }
+func (attributes) GetAPIGroup() string     { return "" }
+func (attributes) GetAPIVersion() string   { return "" }
+func (attributes) IsResourceRequest() bool { return false }
+func (attributes) GetPath() string         { return "" }
```

Comment:
There is this one amazing trick that @s-urbaniak taught me, you can do the following:

```Go
type attributes struct {
  authorizer.Attributes
}
```

and then define only the methods you care about. Delegating on an interface :exploding_head:

Reply:
Nice! Saved me a couple lines. PTAL